ts-client/src/vault/index.ts

```diff
@@ -157,25 +184,146 @@ export default class VaultImpl implements VaultImplementation {
     if (createUserLpTokenIx) {
       preInstructions.push(createUserLpTokenIx);
     }
+
+    return {
+      preInstructions,
+      userToken,
+      userLpToken,
+    };
+  }
+
+  private async createAffiliateATAPreInstructions(owner: PublicKey) {
```

Comment:
We need a function to get partner info
https://github.com/mercurial-finance/vault-periphery/blob/main/affiliate/rust-client/src/partner.rs#L45

Reply:
What does this partner info looks like? Is it this one?
```ts
 {
      "parsed": {
        "info": {
          "isNative": true,
          "mint": "So11111111111111111111111111111111111111112",
          "owner": "7236FoaWTXJyzbfFPZcrzg3tBpPhGiTgXsGWvjwrYfiF",
          "rentExemptReserve": {
            "amount": "2039280",
            "decimals": 9,
            "uiAmount": 0.00203928,
            "uiAmountString": "0.00203928"
          },
          "state": "initialized",
          "tokenAmount": {
            "amount": "0",
            "decimals": 9,
            "uiAmount": 0,
            "uiAmountString": "0"
          }
        },
        "type": "account"
      },
      "program": "spl-token",
      "space": 165
    }
```

I can add this to the demo

Reply:
No it should be like
```
Partner { partner_token: 6EqwccrRKaVqhDijnNvyz9Hh919fsJ2G7JDB9M9HjQuZ, vault: FERjPVNEa7Udq8CEv68h6tPL46Tq7ieE49HrE2wea3XT, total_fee: 2138, fee_ratio: 2000, cummulative_fee: 0 }
```

Reply:
You can refer partner struct in IDL